### PR TITLE
Updated dependencies and migrated to th2 gradle plugin

### PR DESCRIPTION
**Pull request by Nikita-Smirnov-Exactpro**

### CODE REVIEW

**Review by OptimumCode — APPROVED** [review]

**Comment by Nikita-Smirnov-Exactpro** [comment]
I completed my changes, @OptimumCode could you please review to finalize PR

**Review by OptimumCode — APPROVED** [review]

**Review by OptimumCode — CHANGES_REQUESTED** [review]
Something is wrong with maven signing
```
Execution failed for task ':signMavenJavaPublication'.
> Error while evaluating property 'signatory' of task ':signMavenJavaPublication'.
   > Could not read PGP secret key
```

File: .github/workflows/build-release.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+jobs:
+  build:
+    uses: th2-net/.github/.github/workflows/compound-java.yml@main
+    with:
+      build-target: 'Sonatype,Docker'
+      devRelease: false
+      createTag: true
+      docker-username: ${{ github.actor }}
+    secrets:
```
Comment (OptimumCode) [inline .github/workflows/build-release.yml:13]: issue: You should pass signing secrets as well

**Review by OptimumCode — CHANGES_REQUESTED** [review]

**Review by OptimumCode — CHANGES_REQUESTED** [review]
Please, update workflow branch to main

**Review by OptimumCode — APPROVED** [review]

**Review by OptimumCode — APPROVED** [review]

**Pull request merged**